src/module.c

```diff
@@ -1837,6 +1837,20 @@ unsigned long long RM_GetClientId(RedisModuleCtx *ctx) {
     return ctx->client->id;
 }
 
+/* Return the ACL user name used by the client with the specified client ID.
+ * Client ID can be obtained with RM_GetClientId() API. If the client does not
+ * exist, NULL is returned. If the client isn't using an ACL user, an empty
+ * string is returned. */
+RedisModuleString *RM_GetClientUserNameById(RedisModuleCtx *ctx, uint64_t id) {
+    client *client = lookupClientByID(id);
+    if (client == NULL) return NULL;
+
+    sds name = sdsnew(client->user ? client->user->name: "");
```

Comment:
An empty string is a legal username is Redis (it's degenerate, but whatever), I would prefer to set an error code to differentiate between the two cases.

Reply:
How about use `ENODEV` error code? In [acl.c](https://github.com/redis/redis/blob/unstable/src/acl.c#L811). This code is meaning user password is not exist.